relayer/cmd/run/beefy/command.go

```diff
@@ -94,24 +94,20 @@ func run(_ *cobra.Command, _ []string) error {
 			return err
 		}
 
-		err = relay.Start(ctx, eg)
-		if err != nil {
-			logrus.WithError(err).Fatal("Unhandled error")
-			cancel()
+		eg.Go(func() error {
+			err = relay.Start(ctx, eg)
 			return err
-		}
+		})
 	} else {
 		relay, err := beefy.NewOnDemandRelay(&config, keypair)
 		if err != nil {
 			return err
 		}
 
-		err = relay.Start(ctx)
-		if err != nil {
-			logrus.WithError(err).Fatal("Unhandled error")
-			cancel()
+		eg.Go(func() error {
+			err = relay.Start(ctx)
 			return err
-		}
+		})
```

Comment:
Run the on-demand relay in a separate goroutine.

Reply:
The convention with all our relayers is that `Relay.Start` should spawn its own go routines internally.